js/stateless.js/tests/e2e/compress.test.ts

```diff
@@ -0,0 +1,119 @@
+import { describe, it, assert, beforeAll } from 'vitest';
+import {
+    CompressedAccount,
+    CompressedAccountWithMerkleContext,
+    MerkleContext,
+    bn,
+} from '../../src/state';
+import { sendAndConfirmTx, buildAndSignTx } from '../../src/utils';
+
+import { Keypair, Signer } from '@solana/web3.js';
+import { defaultTestStateTreeAccounts } from '../../src/constants';
+import { getTestRpc, newAccountWithLamports } from '../../src/test-utils';
+import { LightSystemProgram, Rpc } from '../../src';
+
+describe('compress', () => {
+    const { merkleTree, nullifierQueue } = defaultTestStateTreeAccounts();
+    let rpc: Rpc;
+    let payer: Signer;
+    let bob: Signer;
+
+    beforeAll(async () => {
+        rpc = await getTestRpc();
+        payer = await newAccountWithLamports(rpc);
+        bob = Keypair.generate();
+    });
+    // Note:
+    // We don't compress SOL yet, therefore cannot spend utxos with value yet.
+    // TODO: add one run with with inputUtxo where lamports: 0
+    it('should compress lamports and then decompress', async () => {
+        const ix = await LightSystemProgram.initCompressedSolPda(
+            payer.publicKey,
+        );
+        const { blockhash: initBlockhash } = await rpc.getLatestBlockhash();
+        const signedInitTx = buildAndSignTx([ix], payer, initBlockhash);
+        await sendAndConfirmTx(rpc, signedInitTx);
+
+        const ixs = await LightSystemProgram.compress({
+            payer: payer.publicKey,
+            address: payer.publicKey,
+            lamports: 20,
+            outputStateTree: merkleTree,
+        });
+
+        /// Send
+        const { blockhash } = await rpc.getLatestBlockhash();
+        const signedTx = buildAndSignTx(ixs, payer, blockhash);
+        await sendAndConfirmTx(rpc, signedTx);
+
+        rpc = await getTestRpc();
+
+        // @ts-ignore
+        const indexedEvents = await rpc.getParsedEvents();
+        assert.equal(indexedEvents.length, 2);
+        assert.equal(indexedEvents[0].inputCompressedAccounts.length, 0);
+        assert.equal(indexedEvents[0].outputCompressedAccounts.length, 1);
+        assert.equal(
+            Number(indexedEvents[0].outputCompressedAccounts[0].lamports),
+            20,
+        );
+
+        assert.equal(
+            indexedEvents[0].outputCompressedAccounts[0].owner.toBase58(),
+            payer.publicKey.toBase58(),
+        );
+
+        assert.equal(indexedEvents[0].outputCompressedAccounts[0].data, null);
```

Comment:
is there an assert the checks that the sender balance has been reduced by the correct amount?